features/readwrite-splitting/core/src/test/java/org/apache/shardingsphere/readwritesplitting/yaml/swapper/YamlReadwriteSplittingRuleConfigurationSwapperTest.java

```diff
@@ -0,0 +1,79 @@
+import org.apache.shardingsphere.readwritesplitting.yaml.config.rule.YamlReadwriteSplittingDataSourceRuleConfiguration;
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Properties;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.Mockito.mock;
+
+public class YamlReadwriteSplittingRuleConfigurationSwapperTest {
+    
+    @Test
+    void assertSwapToYamlConfiguration() {
+        YamlReadwriteSplittingRuleConfiguration actual = getSwapper().swapToYamlConfiguration(creatReadwriteSplittingRuleConfiguration());
+        assertThat(actual.getDataSources().size(), is(1));
+        assertThat(actual.getLoadBalancers().size(), is(1));
+    }
+    
+    private ReadwriteSplittingRuleConfiguration creatReadwriteSplittingRuleConfiguration() {
+        Collection<ReadwriteSplittingDataSourceRuleConfiguration> dataSources = Collections.singletonList(
+                new ReadwriteSplittingDataSourceRuleConfiguration("readwrite", "write_ds", Arrays.asList("read_ds_0", "read_ds_1"), "random"));
+        Map<String, AlgorithmConfiguration> loadBalancers = Collections.singletonMap("myLoadBalancer", new AlgorithmConfiguration("RANDOM", new Properties()));
+        return new ReadwriteSplittingRuleConfiguration(dataSources, loadBalancers);
+    }
+    
+    @Test
+    void assertSwapToObject() {
+        ReadwriteSplittingRuleConfiguration actual = getSwapper().swapToObject(creatYamlReadwriteSplittingRuleConfiguration());
+        assertThat(actual.getDataSources().size(), is(1));
+        assertThat(actual.getLoadBalancers().size(), is(1));
+    }
+    
+    private YamlReadwriteSplittingRuleConfiguration creatYamlReadwriteSplittingRuleConfiguration() {
+        YamlReadwriteSplittingRuleConfiguration result = new YamlReadwriteSplittingRuleConfiguration();
+        YamlReadwriteSplittingDataSourceRuleConfiguration dataSourceRuleConfig = new YamlReadwriteSplittingDataSourceRuleConfiguration();
+        dataSourceRuleConfig.setReadDataSourceNames(Arrays.asList("read_ds_0", "read_ds_1"));
+        dataSourceRuleConfig.setWriteDataSourceName("write_ds");
+        result.getDataSources().put("t_readwrite", dataSourceRuleConfig);
+        YamlAlgorithmConfiguration algorithmConfig = new YamlAlgorithmConfiguration();
+        algorithmConfig.setType("RANDOM");
+        result.getLoadBalancers().put("random_loadbalancer", algorithmConfig);
+        return result;
+    }
+    
+    private YamlReadwriteSplittingRuleConfigurationSwapper getSwapper() {
+        ReadwriteSplittingRuleConfiguration ruleConfig = mock(ReadwriteSplittingRuleConfiguration.class);
+        return (YamlReadwriteSplittingRuleConfigurationSwapper) OrderedSPILoader.getServices(YamlRuleConfigurationSwapper.class, Collections.singleton(ruleConfig)).get(ruleConfig);
```

Comment:
Could we just `new YamlReadwriteSplittingRuleConfigurationSwapper()` to simplify code?